android/beagle/src/main/java/br/com/zup/beagle/android/view/ServerDrivenActivity.kt

```diff
@@ -37,29 +37,41 @@ class ServerDrivenActivity : BeagleActivity() {
     override fun getToolbar(): Toolbar = toolbar
 
     override fun onServerDrivenContainerStateChanged(state: ServerDrivenState) {
-        progress_bar.visibility = if (state is ServerDrivenState.Started) View.VISIBLE else View.GONE
-        handleError(state)
-    }
+        when (state) {
+            is ServerDrivenState.Started -> {
+                showLoading(true)
+                server_driven_container.visibility = View.GONE
+            }
+            is ServerDrivenState.Error -> {
+                if (hasServerDrivenScreen()) {
+                    server_driven_container.visibility = View.VISIBLE
+                    Snackbar.make(server_driven_container, R.string.beagle_label_generic_error, Snackbar.LENGTH_LONG)
+                        .setAction(R.string.beagle_label_try_again) { state.retry() }
+                        .show()
+                } else {
+                    buttonRetry.setOnClickListener {
+                        state.retry()
+                    }
+                    server_driven_container.visibility = View.GONE
+                    include_layout_error.visibility = View.VISIBLE
+                }
```

Comment:
What do you think about a return this code to a function?
```suggestion
                handleError() or showGenericError()
```